include/xtensor/xfft.hpp

```diff
@@ -0,0 +1,294 @@
+#ifndef XFFT_HPP
+#define XFFT_HPP
+
+#include "xmath.hpp"
+#include "xcomplex.hpp"
+#include "xutils.hpp"
+#include "xaxis_slice_iterator.hpp"
+#include "xio.hpp"
+#include "xview.hpp"
+
+namespace xt 
+{
+    namespace fft
+    {
+        template<typename T = double, typename E1, typename E2>
+        xt::xarray<std::complex<T>> fftconvolve(E1&& xvec, E2&& yvec, std::ptrdiff_t axis = -1);
+        
+        namespace detail
+        {
+            template<typename T>
+            T PI()
+            {
+                constexpr double PI = 3.141592653589793238463;
+                return PI;
+            }
+
+            size_t reverseBits(size_t x, int n)
+            {
+                size_t result = 0;
+                for (int i = 0; i < n; i++, x >>= 1)
+                    result = (result << 1) | (x & 1U);
+                return result;
+            }
+
+            bool IsPower2(size_t vecSize) noexcept
+            {
+                return ((vecSize & (vecSize - 1)) == 0);
+            }
+
+            template<typename T, typename E1>
+            auto transformRadix2(const E1& _data)
+            {
+
+                xt::xarray<std::complex<T>> data = _data;
```

Comment:
Same as before. Let's try to use `xtensor` when we can.
Also, isn't the a `xt::cast` for this?

Reply:
I need to make a copy because the algorithm is in place. I switched the signature to a pass by value then do a move at the higher level to avoid unneeded copying.